### PR TITLE
²issues with authentication process

### PR DESCRIPTION
**Pull request by JLTRY**
added reference to _csrf in post

### CODE REVIEW

File: src/dawguk/GarminConnect.php [hunk 1]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
                 $strResponse
             ));
         }
-
+        $doc = new \DOMDocument();
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
        $objDocument = new \DOMDocument();
```

File: src/dawguk/GarminConnect.php [hunk 2]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+        $doc->loadHTML($strResponse);
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
        $objDocument->loadHTML($strResponse);
```

File: src/dawguk/GarminConnect.php [hunk 3]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+        $els = $doc->getElementsByTagName('input');
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
        $arrInputElements = $objDocument->getElementsByTagName('input');
```

File: src/dawguk/GarminConnect.php [hunk 4]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+        for ($i = $els->length; --$i >= 0; ) {
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
        for ($i = $arrInputElements->length; --$i >= 0; ) {
```

File: src/dawguk/GarminConnect.php [hunk 5]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+            $element = $els->item($i);
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
            $objElement = $arrInputElements->item($i);
```

File: src/dawguk/GarminConnect.php [hunk 6]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+            if ($element->getAttribute('name') == '_csrf') {
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
            if ($objElement->getAttribute('name') == '_csrf') {
```

File: src/dawguk/GarminConnect.php [hunk 7]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+            	$csrf= $element->getAttribute('value');
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
            	$strCsrf = $objElement->getAttribute('value'); 
```

File: src/dawguk/GarminConnect.php [hunk 8]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
+        }
         $arrData = array(
             "username" => $strUsername,
             "password" => $strPassword,
-            "_eventId" => "submit",
+            //"_eventId" => "submit",
             "embed" => "true",
-            "displayNameRequired" => "false"
+            //"displayNameRequired" => "false",
+            "_csrf" => $csrf
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]:
```suggestion
            "_csrf" => $strCsrf
```

File: src/dawguk/GarminConnect/Connector.php [hunk 9]
```diff
@@ -103,13 +103,16 @@ public function get($strUrl, $arrParams = array(), $bolAllowRedirects = true)
     * @param bool $bolAllowRedirects
     * @return mixed
     */
-    public function post($strUrl, $arrParams = array(), $arrData = array(), $bolAllowRedirects = true)
+    public function post($strUrl, $arrParams = array(), $arrData = array(), $referer = null, $bolAllowRedirects = true)
```
Comment (dawguk) [inline src/dawguk/GarminConnect/Connector.php]:
```suggestion
    public function post($strUrl, $arrParams = array(), $arrData = array(), $strReferer = null, $bolAllowRedirects = true)
```

File: src/dawguk/GarminConnect/Connector.php [hunk 10]
```diff
@@ -103,13 +103,16 @@ public function get($strUrl, $arrParams = array(), $bolAllowRedirects = true)
...
     {
 
         curl_setopt($this->objCurl, CURLOPT_HEADER, true);
         curl_setopt($this->objCurl, CURLOPT_FRESH_CONNECT, true);
         curl_setopt($this->objCurl, CURLOPT_FOLLOWLOCATION, (bool)$bolAllowRedirects);
         curl_setopt($this->objCurl, CURLOPT_CUSTOMREQUEST, "POST");
+	    if ($referer != null) {
```
Comment (dawguk) [inline src/dawguk/GarminConnect/Connector.php]:
```suggestion
	    if ($strReferer != null) {
```

File: src/dawguk/GarminConnect/Connector.php [hunk 11]
```diff
@@ -103,13 +103,16 @@ public function get($strUrl, $arrParams = array(), $bolAllowRedirects = true)
...
+			 curl_setopt($this->objCurl, CURLOPT_REFERER, $referer);
```
Comment (dawguk) [inline src/dawguk/GarminConnect/Connector.php]:
```suggestion
			 curl_setopt($this->objCurl, CURLOPT_REFERER, $strReferer);
```

File: src/dawguk/GarminConnect/Connector.php (hunk 10 above)
Comment (dawguk) [inline src/dawguk/GarminConnect/Connector.php]: Indenting on this if statement needs correcting

File: src/dawguk/GarminConnect.php [hunk 12]
```diff
@@ -121,16 +141,25 @@ private function authorize($strUsername, $strPassword)
...
         );
 
-        $strResponse = $this->objConnector->post("https://sso.garmin.com/sso/login", $arrParams, $arrData, false);
+        $strResponse = $this->objConnector->post("https://sso.garmin.com/sso/login", $arrParams, $arrData, $req, false);
```
Comment (dawguk) [inline src/dawguk/GarminConnect.php]: I can't see what `$req` refers to - it doesn't appear to have been initialized.

**Comment by JLTRY** [comment]
remarks taken into account

**Comment by JLTRY** [comment]
done in new push

**Pull request closed without merging**